File: 2024-dgu-spring-camp/11/398.cpp
// 수 찾기
// https://study.helloalgo.co.kr/study/1019/room/598/6553/source/828365/

#include <iostream>
#include <vector>
using namespace std;

vector<int> nums, targets;

bool findNum(int n, int target)
{
    int start = 0, end = n - 1;
    while (start <= end)
    {
        int mid = (start + end) / 2;

        if (nums[mid] == target)
        {
            return true;
        }
        else if (nums[mid] > target)
        {
            end = mid - 1;
        }
        else
        {
            start = mid + 1;
        }
    }
    return false;
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int N, M;
    cin >> N >> M;

    int temp;
    for (int i = 0; i < N; i++)
    {
        cin >> temp;
        nums.push_back(temp);
    }
    for (int i = 0; i < M; i++)
    {
        cin >> temp;
        targets.push_back(temp);
    }

    for (int target : targets)
    {
        if (findNum(N, target))
        {
            cout << "1\n";
        }
        else
        {
            cout << "0\n";
        }
    }
}

File: 2024-dgu-spring-camp/11/398.py
# 수 찾기
# https://study.helloalgo.co.kr/study/1019/room/598/6553

N, M = map(int, input().split())
nums = list(map(int, input().split()))
targets = list(map(int, input().split()))

def findNum(n, arr, target):
    start, end = 0, n-1
    answer = None
    while start <= end:
        mid = (start + end)//2
        
        if arr[mid] == target:
            answer = mid
            return True
        elif arr[mid] > target:
            end = mid-1
        else:
            start = mid+1
            
    return False

for target in targets:
    print('1' if findNum(N, nums, target) else '0')
